### PR TITLE
Update to Alpine 3.20

### PR DESCRIPTION
**Pull request by yosifkit**
Remove EOL `2.0` images. Remove Alpine from the `2.2` image since it cannot be built on a supported release of Alpine now that 3.16 is end of life.

Related https://github.com/docker-library/official-images/pull/16801

### CODE REVIEW

File: versions.sh [hunk 1]
```diff
@@ -39,10 +37,10 @@ for version in "${versions[@]}"; do
 				debian: env.debian,
 				alpine: env.alpine,
 			}
+			| if env.version == "2.2" then del(.alpine) else . end
```
Comment (tianon) [inline versions.sh:40]: Any particular reason not to put the comment directly here in the `jq` instead? :sweat_smile:

(maybe even match on `.version | startswith("2.2.")` so we can avoid `export version` at all?)

Comment (tianon) [reply versions.sh:40]: Oh, we use `export version` further down -- in that case I'd actually move it even further up to the first line of the `for` (right after it's definitely got the value we expect).

**Review by tianon — APPROVED** [review]

**Comment by TimWolla** [comment]
Note: Technically HAProxy 2.0 is not EOL yet. See: https://github.com/docker-library/haproxy/pull/228#discussion_r1554623341

**Comment by tianon** [comment]
Doh, yep, we should bring that back (and only drop the Alpine variants of it).

**Comment by tianon** [comment]
(Thank you!)

**Pull request merged**